connector.go

```diff
@@ -331,6 +331,11 @@ func NewConnector(hostConfig HostConfig, transportConfig TransportConfig,
 	connector.Requestor.Init(connector.TransportConfig)
 
 	res = connector
-
+	// GET UserProfile request is used here to validate connector's basic auth and reachability.
+	userprofile := NewUserProfile(UserProfile{})
+	_, err = connector.makeRequest(GET, userprofile, "")
```

Comment:
Suggestion: if possible, please use the GetObject() method instead. 

Question:  So, we just need to check that the request is successful? We don't need to check the result?

Reply:
@yuewko I am validating two things as below:
1) reachability means verifying if ip and port is reachable 
2) basic authentication means username and password.

For Suggestion: Ok I will do it.

Reply:
@yuewko looking at GetObject I feel we dont need to use it in this case, as we donot care about the response of UserProfile.

Reply:
`makeRequest()` is meant to be a service method used by Get/Update/DeleteObject() methods.. For maintainability, we should try to minimize use of service methods.
BTW, I understand that we don't need the result, but the overhead is minimal since we are only doing this once per connection.

I don't have strong opinion on this one. I'll approve it.

Reply:
Ok Ko i will do the changes and will consider this from next commits.